### PR TITLE
[FL-3833] Furi: event loop

### PR DESCRIPTION
**Pull request by skotopes**
# What's new

- Furi: event loop
- Furi: message queue event loop integration
- ViewDispatcher: multiple queue + event loop

# Implemented

- EventLoop foundation: level based In/Out.
- Message Queue integration
- Lazy ticks integration
- Postponed events
- Obect to Epoll abstraction layer

# Missing 

- Edge events
- One Shot and other less common flags
- Priorities
- Priority inversion

# Verification 

- Full set of tests

# Checklist (For Reviewer)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

**Comment by CookiePLMonster** [comment]
Can epoll events add/remove message queues to/from that epoll or is it not allowed?

**Comment by skotopes** [comment]
@CookiePLMonster yes, that should be possible. However I have not yet covered it with tests.

**Comment by CookiePLMonster** [comment]
@skotopes the reason I'm asking because literally the first thing that came to mind when I studied this PR was "Does this mean the scenes in a scene manager can now easily have their own message queues if needed?" which naturally means queues would be added and removed from a view dispatcher call (I think?).

**Comment by skotopes** [comment]
@CookiePLMonster yes. That was original intent of this PR: one event loop to rule them all.

File: furi/core/event_loop.h [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+ */
+
+/** Anonymous message queue type */
+typedef struct FuriMessageQueue FuriMessageQueue;
+
+/** Callback type for message queue
+ *
+ * @param      queue    The queue that triggered event
+ * @param      context  The context that was provided on
+ *                      furi_event_loop_message_queue_add call
```
Comment (CookiePLMonster) [inline furi/core/event_loop.h]: There is no `furi_event_loop_message_queue_add` in the current public API.

**Comment by CookiePLMonster** [comment]
Now that the PR is out of draft, question about the usage - if `furi_event_loop_message_queue_subscribe` mandates the event loop must be idle for subscriptions/unsubscriptions to work, how does one add/remove message queues during the lifetime of the event loop? For example when switching scenes.

**Comment by skotopes** [comment]
@CookiePLMonster this feature coming in next release. This PR is a first part of many )

File: furi/core/event_loop.h (hunk 1 above)
Comment (skotopes) [reply furi/core/event_loop.h]: Fixed

**Review by gsurkov — DISMISSED** [review]

**Review by gsurkov — APPROVED** [review]

**Pull request merged**